pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1629,14 +1629,21 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     if (PyLong_Check(obj)) {
         PRINTMARK();
         tc->type = JT_LONG;
-        GET_TC(tc)->longValue = PyLong_AsLongLong(obj);
+        int overflow = 0;
+        GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, &overflow);
+        _Bool err;
+        err = (GET_TC(tc)->longValue == -1) && PyErr_Occurred();
 
-        exc = PyErr_Occurred();
+        if (overflow && !err){
```

Comment:
```suggestion
        if (overflow){
```

I think from the doctoring overflow would only get set when there is no error, so don't need both